commands/watch.go

```diff
@@ -25,13 +26,23 @@ func (command *WatchCommand) Execute(args []string) error {
 		return err
 	}
 
+	var buildId int
 	client := target.Client()
-	build, err := GetBuild(client, target.Team(), command.Job.JobName, command.Build, command.Job.PipelineName)
-	if err != nil {
-		return err
+	if command.Job.JobName != "" || command.Build == "" {
+		build, err := GetBuild(client, target.Team(), command.Job.JobName, command.Build, command.Job.PipelineName)
+		if err != nil {
+			return err
+		}
+		buildId = build.ID
+	} else if command.Build != "" {
+		buildId, err = strconv.Atoi(command.Build)
+
+		if err != nil {
+			return err
+		}
 	}
 
-	eventSource, err := client.BuildEvents(fmt.Sprintf("%d", build.ID))
+	eventSource, err := client.BuildEvents(fmt.Sprintf("%d", buildId))
```

Comment:
A bit odd that this is a string, especially since it's converted to and from in the lines above. But that'd mean a change in `go-concourse`, so I won't block this on that change.